AmazonAdvertisingApi/Client.php

```diff
@@ -385,14 +763,18 @@ public function getSnapshot($snapshotId)
         return $req;
     }
 
-    public function requestReport($recordType, $data = null)
+    public function requestSponsoredProductsReport($recordType, $data = null)
     {
-        return $this->_operation("{$recordType}/report", $data, "POST");
+        return $this->_operation(
+            "{$this->prefixes["products"]}{$recordType}/report", $data, "POST"
+        );
     }
 
     public function getReport($reportId)
     {
-        $req = $this->_operation("reports/{$reportId}");
+        $req = $this->_operation(
+            "{$this->prefixes["report"]}reports/{$reportId}"
+        );
         if ($req["success"]) {
```

Comment:
Would this be more readable if _operation() took the string used to fetch the prefix as a parameter and called the $this->prefixes itself.

ie so you only have to call
```
$this->_operation(
            "reports/{$reportId}",
            "report"
        );
```

Reply:
yes, I much prefer this.